### PR TITLE
Deprecate TaskInputs query methods

### PR DESCRIPTION
**Pull request by lptr**
These methods expose internal state that is expensive to calculate. Instead of using them, user code can rely on declared task properties.

### CODE REVIEW

**Comment by lptr** [comment]
Feedback build: https://builds.gradle.org/viewQueued.html?itemId=9895234

File: subprojects/docs/src/docs/release/notes.md [hunk 1]
```diff
@@ -80,6 +80,15 @@ The following are the newly deprecated items in this Gradle release. If you have
 
 The command line options for searching in parent directories for a `settings.gradle` file (`-u`/`--no-search-upward`) has been deprecated and will be removed in Gradle 5.0. A Gradle project should always define a `settings.gradle` file to avoid the performance overhead of walking parent directories.
 
+### Deprecation of `TaskInputs` and `TaskOutputs` methods
```
Comment (wolfs) [inline subprojects/docs/src/docs/release/notes.md]: How about "Deprecation of `TaskInputs` and `TaskOutputs` query methods"?

**Comment by lptr** [comment]
Feedback build: https://builds.gradle.org/viewQueued.html?itemId=9895444

**Comment by lptr** [comment]
Feedback build: https://builds.gradle.org/viewQueued.html?itemId=9899595

**Comment by lptr** [comment]
Build: https://builds.gradle.org/viewQueued.html?itemId=9900584

**Pull request merged**